### PR TITLE
Add beekeeping and beekeeping acessories

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Allow players to keep bees"

#### Purpose of change
Add something like #57355, and add some bee-related things.

#### Describe the solution
Add wild honey bee hives, which can be harvested for honey, wax, and their queen. Using the queen, players can create primers which are used to seed their own bee colonies in apiaries, and extract honey from them seasonally, providing a renewable source of the substance for private use and/or profit.

#### Describe alternatives you've considered
Removing the primers and requiring direct culture in a hive, not making queen bees an item, or not adding normal-sized bees. Using an EOC-based function instead of `harvest_ter`.

#### Testing
I tested everything multiple times, and it all works well.

#### Additional context
With the current, non-EOC system, honey scrapers do not serve a role in this process, as I took the "simpler" route to implementation. You also don't need beekeeping gear to extract honey from hives at the present time, although many beekeepers don't use these things.

### CODE REVIEW

**Comment by Karol1223** [comment]
To be honest, there probably should either be a chance for them to randomly mutate or an actual in-game way to, correctly from the lore side, prevent mutations in farmed animals.

**Comment by Grubleafeater** [comment]
> To be honest, there probably should either be a chance for them to randomly mutate or an actual in-game way to, correctly from the lore side, prevent mutations in farmed animals.

Do you want CDDAForestry? Because that's how you get CDDAForestry.

**Comment by MNG-cataclysm** [comment]
> Do you want CDDAForestry? Because that's how you get CDDAForestry.

Is that a mod?

**Comment by Karol1223** [comment]
> Do you want CDDAForestry? Because that's how you get CDDAForestry.

Beekeeping and similarly keeping fish in ponds is something I had on my mind for ages and always avoided it like fire because I didn't want to do it in such a way that it didn't tie into wildlife mutations at all. I asked about ways to avoid mutations in kept wildlife on the discord and I never got an answer. You seem to have done a fairly good job with this PR, it's just that it's missing the crucial aspect connected to insects in CDDA universe and I don't have any wisdom to share on how to tie it.

**Comment by MNG-cataclysm** [comment]
> Beekeeping and similarly keeping fish in ponds is something I had on my mind for ages and always avoided it like fire because I didn't want to do it in such a way that it didn't tie into wildlife mutations at all. I asked about ways to avoid mutations in kept wildlife on the discord and I never got an answer. You seem to have done a fairly good job with this PR, it's just that it's missing the crucial aspect connected to insects in CDDA universe and I don't have any wisdom to share on how to tie it.

I suppose a giant bee could randomly spawn from one of the hives, possibly friendly to the player. I think field emissions can spawn monsters, if I remember correctly.

**Comment by Karol1223** [comment]
> I suppose a giant bee could randomly spawn from one of the hives, possibly friendly to the player. I think field emissions can spawn monsters, if I remember correctly.

I don't know if friendly is viable. @Venera3 would have to comment on this here.

**Comment by Fris0uman** [comment]
Can you stick in a basement and have it produce honey?

**Comment by MNG-cataclysm** [comment]
> Can you stick in a basement and have it produce honey?

With the current implementation, yes. I don’t know how I could get it to check for a suitable environment around it.

**Review by Venera3 — CHANGES_REQUESTED** [review]
This is not really how bees (or beekeeping) works. Having fire-and-forget apiaries is fine, but 
- you're not getting 225 *litres* of honey out of it, *especially* not for half a year - a well managed small but established hive will get you ~30ish kg per year
- you're not getting extra queens - a beehive has a single queen, for "she kills other queens-to-be(e)" reasons, you don't get to handwave that away even if it makes it harder to set up. The way extra queens happen is letting a colony swarm, or capturing a "wild" swarm.

Things you can do that would make sense:
- Set up rare bee tree versions of trees you can harvest in the autumn, turning them into the non-beed tree and yielding a few dozen kgs of honey+wax. Plus points if you're a Bear.
- If you *want* to get an apiary started you could squeeze in finding wild swarms in the spring while foraging (way too broad a timeframe, but atm harvest are per season only). No yoinking queens from existing hives or magic de novo queens of survivor comfort. This is a considerable stretch but I'd personally let it go through.
- Harvest in the autumn only and a short rot time on the swarm items would ensure that the timeframe of whole thing is not fantastically short.
- Reign in the yields to "average for a single-box hive" territory. With the complete lack of beekeeping you're doing even opening the thing is a stretch, but it works as a minimal viable product as long as the yields are small.
- If you want to do it properly , it would be much cleaner with furniture transforming on a timer, but that's a code change that might get messy. As an upside you could have "breeder" hives you "extract" swarms from in return for no/reduced honey yields (as a surrogate of letting/inducing swarming and capturing the result)

The mutant question is not really relevant for now, to be honest - I work with "not every mutant is noticable" as a headcanon, so normal-ish bees sticking around a few years is all right.

Random nitpicks 
- modern bee boxes haven't been used for "millenia" - frame hives were invented in 1852, previous apiaries used disposable hives you destroyed to get at the good stuff (or bee trees outfitted with hinges, but that's a very particular tradition)
- Beehives aren't round unless you're stuck in the Hundred Acre Woods. Bees nest in tree cavities or similar enclosed spaces, and if they *have* to nest without one they make parallell hanging layers of wax [like so](https://i.pinimg.com/originals/a4/e8/bd/a4e8bd577424767101fae00ac2ff2f47.jpg) , but that is a death sentence for the poor girls outside of the tropics.

**Comment by MNG-cataclysm** [comment]
By “~30is kg”, do you mean that of honey, or for honey and comb included?

**Comment by MNG-cataclysm** [comment]
@Venera3 I implemented most of the changes you requested, though I don't know how I'd get the terrain to change on an individual timer. Do the hive "starter" items make any sense, or should the queen be directly included within the apiary recipe?

File: data/json/items/resources/misc.json [hunk 1]
```diff
@@ -282,5 +282,50 @@
     "material": [ "rubber", "steel" ],
     "symbol": "o",
     "color": "dark_gray"
+  },
+  {
+    "id": "hive_starter_immature",
```
Comment (Venera3) [inline data/json/items/resources/misc.json:287]: You don't need the two-step process here - ideally you rehome a found swarm directly, and they don't start pumping out workers without having found an appropriate home (that a sealed glass jar very much isn't). If the swarm works as a liquid you can just use that in the construction recipe.

**Review by Venera3 — CHANGES_REQUESTED** [review]
Better, but not quite there yet. I'll poke at it some more later.

File: data/json/items/comestibles/other.json [hunk 2]
```diff
@@ -1085,5 +1085,20 @@
     "fun": -15,
     "freezing_point": -274,
     "flags": [ "NUTRIENT_OVERRIDE" ]
+  },
+  {
+    "id": "queen_bee",
```
[the remark this answers is not in the document]
Comment (MNG-cataclysm) [reply data/json/items/comestibles/other.json:1090]: I tested it, and foraging cannot handle liquids.

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 3]
```diff
@@ -1811,6 +1811,79 @@
...
+    "id": "t_beehive_natural",
+    "name": "bee hive",
+    "description": "A hive of bees located in a cavity inside a tree, swarming in and out at a feverish pace.  There's probably honey inside.",
+    "symbol": "h",
+    "color": [ "yellow" ],
+    "move_cost": 0,
+    "coverage": 10,
+    "flags": [ "FLAMMABLE_ASH", "NOITEM" ],
+    "examine_action": "harvest_ter",
+    "harvest_by_season": [ { "seasons": [ "spring", "summer" ], "id": "beehive_harv" } ],
```
Comment (Venera3) [inline data/json/furniture_and_terrain/terrain-flora.json:1825]:
```suggestion
    "harvest_by_season": [ { "seasons": [ "summer", "autumn" ], "id": "beehive_harv" } ],
```
You're stealing their winter stockpile(you monster), in the spring they barely have any honey left.

File: data/json/items/comestibles/other.json [hunk 4]
```diff
@@ -1086,6 +1086,22 @@
...
+    "type": "COMESTIBLE",
+    "comestible_type": "FOOD",
+    "name": { "str": "swarm of bees", "str_pl": "swarms of bees" },
+    "description": "A swarm of honey bees, displaced from the hive.  Without support and nutrients, or taking root somewhere, they will die in a few days.",
+    "//": "While bees have mass and volume, it is easier to round to zero for the purposes of the game.",
+    "price": 0,
+    "price_postapoc": 0,
+    "spoils_in": "2 days",
+    "flags": [ "ZERO_WEIGHT", "INEDIBLE" ],
+    "volume": "500 ml",
```
Comment (Venera3) [inline data/json/items/comestibles/other.json:1100]: Spilled beeswarms being lost notwithstanding I think it really should work as a liquid for our purposes unless liquid forage results derp out.

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 5]
```diff
@@ -1811,6 +1811,79 @@
...
+    "name": "apiary",
+    "description": "A man-made wooden box designed to hold honeybees and facilitate the care for a hive.  The modular construction allows for inspecting the bees and harvesting their bounty without destroying the hive completely.  This year's harvest should be ready by now.",
+    "symbol": "h",
+    "color": [ "yellow" ],
+    "move_cost": 1,
+    "coverage": 50,
+    "flags": [ "FLAMMABLE_ASH", "NOITEM" ],
+    "examine_action": "harvest_ter",
+    "//": "While this does give a queen bee, handwave it as an exrta to facilitate the development of new hives.",
+    "harvest_by_season": [ { "seasons": [ "summer" ], "id": "beehive_harv" } ],
```
Comment (Venera3) [inline data/json/furniture_and_terrain/terrain-flora.json:1849]:
```suggestion
    "harvest_by_season": [ { "seasons": [ "autumn" ], "id": "beehive_harv" } ],
```
Unlike the bee trees these are not established colonies, so you have to wait for your honey.

**Review by Venera3 — CHANGES_REQUESTED** [review]
Some minor adjustments, but almost good to go.

File: data/json/items/comestibles/other.json (hunk 4 above)
Comment (MNG-cataclysm) [reply data/json/items/comestibles/other.json:1100]: Forage results derp out when this is set as a liquid. It instantly drops to the ground, becomes "dirty", and leaves the player with no way to pick it up.

Comment (Venera3) [reply data/json/items/comestibles/other.json:1100]: Okay, weirdly complacent swarms it is I suppose. Might be worth a feature request for future reference.

**Review by Venera3 — APPROVED** [review]

**Comment by Karol1223** [comment]
So... ``t_beehive_natural`` doesn't spawn anywhere now, does it?

**Comment by oosyrag** [comment]
Was it ever confirmed that empty apiaries ever transformed back into harvestable ones? Just wanted to check I had the right idea for #68910, or if it is actually working the way it is now and something else is wrong in the reporter's game.

**Pull request merged**